### ISSUE TITLE
Change some external dependency version requirements to any - at least for http package

### ISSUE THREAD

**Issue by navispatial** [navibyte/geospatial#247]
As noted in #241 some external dependency version requirements may cause problems.

See the official [documentation](https://dart.dev/tools/pub/dependencies) about package dependencies.

Latest published packages has following requirements on `pubspec.yaml` files.

**geobase** (version 1.3.0)
```yaml
environment:
  sdk: '>=2.17.0 <4.0.0'

dependencies:
  collection: ^1.18.0
  equatable: ^2.0.0
  meta: ^1.7.0
  proj4dart: ^2.0.0
```

**geodata** (version 1.2.1)
```yaml
environment:
  sdk: '>=2.17.0 <4.0.0'

dependencies:
  equatable: ^2.0.0
  http: ^0.13.0
  meta: ^1.7.0
  geobase: ^1.1.0
```

Analyzing current (2024-11-04) versions of these packages:

Package | Latest version | Requirement here | Latest that resolves when testing package using `dart pub upgrade`)
-------- | -------------- | ------------------ | ------------------------------------
collection | 1.19.1 (min sdk 3.4) | ^1.18.0 | 1.18.0
equatable | 2.0.5 (min sdk 2.12) | ^2.0.0 | 2.0.5
meta | 1.16.0 (min sdk 2.12) | ^1.7.0 | 1.16.0
proj4dart | 2.1.0 (min sdk 2.12) | ^2.0.0 | 2.1.0
http | 1.2.2 (min sdk 3.3) | ^0.13.0 | 0.13.6

The biggest problem is maybe `http` package, with dependency requirement `^0.13.0` it's impossible to resolve to next major versions 1.0.0+ (like 1.2.2) of it even if some other packages might require it.

Tasks here:
* analyze which dependency requirements are enough for `geobase` and `geodata` (or could `any` be used) so that other packages won't have conflicting requirements on same external libraries
* update dependencies on next versions 1.4.0 coming November 2024

Another note: `equatable` is used only in very few places (at least in `geobase`) and use of it could be refactored out (even if it's not currently causing any dependency problems).

**Comment by navispatial** [comment]
Testing following settings.

**geobase**

The `equatable` package was removed totally (see #248), other versions kept same as before.

```yaml
environment:
  sdk: '>=2.17.0 <4.0.0'

dependencies:
  collection: ^1.18.0
  meta: ^1.7.0
  proj4dart: ^2.0.0
```

**geodata**

The `equatable` package cannot easily to be removed, so keep it here.

But set version of it and the `http` package to `any`.

```yaml
environment:
  sdk: '>=2.17.0 <4.0.0'

dependencies:
  equatable: any
  http: any
  meta: ^1.7.0
  geobase: ^1.1.0
```

When running `dart pub downgrade` and then running local tests with `dart test` all test fail. Seems dart tool related thing however, tests cannot be created. Some version conflict between test program, analyzer, io, file and/or semver package (or what ever....). Maybe older Dart SDK should be used, but not available on development machine, so just test with latest Dart SDK.

When running `dart pub downgrade` and calling example command line programs they succeed all.

When running `dart pub upgrade` both local tests and example programs succeed.

With Dart SDK 3.5.4 dependency constraints above resolve to 

geobase:
```
├── collection 1.19.1
├── meta 1.16.0
├── proj4dart 2.1.0
```

geodata:
```
├── equatable 2.0.5
├── http 1.2.2
├── meta 1.16.0
```

**Comment by navispatial** [comment]
Maybe using `any` dependency to `equatable` on the `geodata` package is NOT very good idea, see discussion on #248.

**Comment by navispatial** [comment]
An interesting article about handling dependencies:

https://apps.theodo.com/article/how-to-handle-breaking-changes-in-dart-sdk-dependencies

**Comment by navispatial** [comment]
For the published release 2024-11-10 the following constraints finally were used.

`geobase`

```yaml
environment:
  sdk: '>=2.17.0 <4.0.0'

dependencies:
  meta: ^1.7.0
  proj4dart: ^2.0.0
```

`geodata`

```yaml
environment:
  sdk: '>=2.17.0 <4.0.0'

dependencies:
  http: '>=0.13.0 <2.0.0'
  meta: ^1.7.0
  geobase: ^1.1.0
```

**Comment by navispatial** [comment]
Implemented in [geobase version 1.4.0](https://pub.dev/packages/geobase/versions/1.4.0+1) and [geodata version 1.3.0](https://pub.dev/packages/geodata/versions/1.3.0), see [milestone](https://github.com/navibyte/geospatial/milestone/6).
